swarm of fails in datalad/git-annex master branch testing

see https://github.com/datalad/git-annex/runs/2881114114?check_suite_focus=true  not yet sure what/why since we were green for https://github.com/datalad/datalad/pull/5691 PR

<details>
<summary>partial extract</summary> 

```shell
======================================================================
FAIL: datalad.support.tests.test_external_versions.test_external_versions_rogue_module
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_external_versions.py", line 141, in test_external_versions_rogue_module
    assert_in('pickaboo', cml.out)
Warning: Error: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
-------------------- >> begin captured logging << --------------------
datalad: WARNING: Failed to import module verycustomrogue__ due to 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.support.tests.test_github_.test__make_github_repos
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_github_.py", line 113, in test__make_github_repos
    assert_in('Failed to create repository while using token 1to...: very bad status', cml.out)
Warning: Error: 'Failed to create repository while using token 1to...: very bad status' not found in '[WARNING] Failed to create repository while using token 1to...: \n'
-------------------- >> begin captured logging << --------------------
datalad.support.github_: WARNING: Failed to create repository while using token 1to...: 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.support.tests.test_sshconnector.test_ssh_manager_close_no_throw
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 739, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/support/tests/test_sshconnector.py", line 202, in test_ssh_manager_close_no_throw
    assert_in('Failed to close a connection: oh I am so bad', cml.out)
AssertionError: 'Failed to close a connection: oh I am so bad' not found in '[DEBUG] Closing 1 SSH connections...\n[DEBUG] Failed to close a connection: \n'
-------------------- >> begin captured logging << --------------------
datalad.support.sshconnector: DEBUG: Closing 1 SSH connections...
datalad.support.sshconnector: DEBUG: Failed to close a connection: 
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: datalad.tests.test_utils.test_import_modules_fail
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/test_utils.py", line 1098, in test_import_modules_fail
    ok_startswith(failures[0], "Fail datalad.bogus: No module")
  File "/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/datalad/tests/utils.py", line 435, in ok_startswith
    msg="String %r doesn't start with %r" % (s, prefix))
AssertionError: String 'Fail datalad.bogus: ' doesn't start with 'Fail datalad.bogus: No module'

```
</details>

full logs could be found on smaug:/mnt/datasets/datalad/ci/git-annex/builds/2021/06/cron-20210622/build-ubuntu.yaml-317-a1f634dc-failed and I also see lots of hits for the 'pickaboo' in the logs for #5691:

```shell
(tinuous-dev) datalad@smaug:/mnt/datasets/datalad/ci/logs/2021/06/08/pr/5691$ git grep 'pickaboo'
3dcf2d3/github-Test on macOS-1425-failed/1_test (brew).txt:2021-06-08T12:51:55.5946970Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/1_test (brew).txt:2021-06-08T12:51:55.5948100Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/2_test (snapshot).txt:2021-06-08T13:01:34.0297150Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/2_test (snapshot).txt:2021-06-08T13:01:34.0298580Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/test (brew)/8_Run tests.txt:2021-06-08T12:51:55.5946960Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/test (brew)/8_Run tests.txt:2021-06-08T12:51:55.5948090Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
3dcf2d3/github-Test on macOS-1425-failed/test (snapshot)/8_Run tests.txt:2021-06-08T13:01:34.0297130Z     assert_in('pickaboo', cml.out)
3dcf2d3/github-Test on macOS-1425-failed/test (snapshot)/8_Run tests.txt:2021-06-08T13:01:34.0298570Z AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/57k0ylu8v3p39hqb.txt:[00:37:34]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/57k0ylu8v3p39hqb.txt:[00:37:34] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/fvlsmm3k5y2poi8v.txt:[00:44:40]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/fvlsmm3k5y2poi8v.txt:[00:44:40] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/jnnx886qnxc0j47q.txt:[00:44:50]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/jnnx886qnxc0j47q.txt:[00:44:50] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/n409j3f11lr50im0.txt:[00:32:40]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/n409j3f11lr50im0.txt:[00:32:40] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
588644c/appveyor-6107-failed/p28vcoblx377qwwi.txt:[00:43:45]     assert_in('pickaboo', cml.out)
588644c/appveyor-6107-failed/p28vcoblx377qwwi.txt:[00:43:45] AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/1.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/1.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/2.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/2.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/3.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/3.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/6.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/6.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/7.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/7.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
de982c5/travis-12020-failed/8.txt:    assert_in('pickaboo', cml.out)
de982c5/travis-12020-failed/8.txt:AssertionError: 'pickaboo' not found in '[WARNING] Failed to import module verycustomrogue__ due to \n'
```
so I guess it was not that clean

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -51,8 +51,6 @@ shallow_clone: true
 
 environment:
   DATALAD_TESTS_SSH: 1
-  # The following is currently necessary as we test a lot of logging output in high detail:
-  DATALAD_LOG_EXC: 1
 
   # Do not use `image` as a matrix dimension, to have fine-grained control over
   # what tests run on which platform

--- a/.github/workflows/test_crippled.yml
+++ b/.github/workflows/test_crippled.yml
@@ -6,9 +6,6 @@ jobs:
   test:
 
     runs-on: ubuntu-latest
-    env:
-      # The following is currently necessary as we test a lot of logging output in high detail:
-      DATALAD_LOG_EXC: 1
     steps:
     - name: Set up system
       shell: bash

--- a/.github/workflows/test_macos.yml
+++ b/.github/workflows/test_macos.yml
@@ -7,9 +7,6 @@ on:
 jobs:
   test:
     runs-on: macos-latest
-    env:
-      # The following is currently necessary as we test a lot of logging output in high detail:
-      DATALAD_LOG_EXC: 1
     strategy:
       fail-fast: false
       matrix:

--- a/.github/workflows/test_win2019_disabled
+++ b/.github/workflows/test_win2019_disabled
@@ -6,9 +6,6 @@ jobs:
   test:
 
     runs-on: windows-2019
-    env:
-      # The following is currently necessary as we test a lot of logging output in high detail:
-      DATALAD_LOG_EXC: 1
 
     strategy:
       fail-fast: false

--- a/.travis.yml
+++ b/.travis.yml
@@ -27,8 +27,6 @@ env:
     # How/which git-annex we install.  conda's build would be the fastest, but it must not
     # get ahead in PATH to not shadow travis' python
     - _DL_ANNEX_INSTALL_SCENARIO="miniconda --batch git-annex=8.20201007 -m conda"
-    # The following is currently necessary as we test a lot of logging output in high detail:
-    - DATALAD_LOG_EXC=1
 
 matrix:
   include:

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -135,6 +135,7 @@ def setup_package():
     # own HOME where we pre-setup git for testing (name, email)
     if 'GIT_HOME' in os.environ:
         set_envvar('HOME', os.environ['GIT_HOME'])
+        set_envvar('DATALAD_LOG_EXC', "1")
     else:
         # we setup our own new HOME, the BEST and HUGE one
         from datalad.utils import make_tempfile

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -149,6 +149,8 @@ def setup_package():
 [user]
 	name = DataLad Tester
 	email = test@example.com
+[datalad "log"]
+	exc = 1
 """)
         _TEMP_PATHS_GENERATED.append(new_home)
 

--- a/datalad/__init__.py
+++ b/datalad/__init__.py
@@ -103,6 +103,8 @@ test_http_server = None
 def setup_package():
     import os
     from datalad.utils import on_osx
+    from datalad.tests import _TEMP_PATHS_GENERATED
+
     if on_osx:
         # enforce honoring TMPDIR (see gh-5307)
         import tempfile
@@ -136,7 +138,6 @@ def setup_package():
     else:
         # we setup our own new HOME, the BEST and HUGE one
         from datalad.utils import make_tempfile
-        from datalad.tests import _TEMP_PATHS_GENERATED
         # TODO: split into a function + context manager
         with make_tempfile(mkdir=True) as new_home:
             pass

